src/ansys/additive/core/parametric_study/parametric_study.py

```diff
@@ -1926,3 +1923,73 @@ def _validate_microstructure_input(
             return (True, "")
         except ValueError as e:
             return (False, (f"Invalid parameter combination: {e}"))
+
+    @staticmethod
+    def _filter_dataframe(
+        df: pd.DataFrame,
+        simulation_ids: list[str] = None,
+        type: list[SimulationType] = None,
+        priority: int = None,
+        iteration: int = None,
+    ) -> pd.DataFrame:
+        """Apply filters to the parametric study data frame.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Parametric study data frame to filter.
+        simulation_ids: list[str], default: None
+            List of simulation IDs to include. The default is ``None``, in which case
+            all simulations with status of :obj:`SimulationStatus.NEW` are selected.
+        type : list, default: None
+            List of simulation types to include. The default is ``None``, in which case
+            all simulation types are selected.
+        priority : int, default: None
+            Priority of simulations to include. The default is ``None``, in which case
+            all priorities are selected.
+        iteration : int, default: None
+            Iteration number of simulations to include. The default is ``None``, in which case
+            all iterations are selected.
+
+        Returns
+        -------
+        pd.DataFrame
+            Filtered view of the parametric study data frame
+        """
+
+        # Initialize the filtered view of the data frame
+        view = df
```

Comment:
I'm a little confused by this. `=` doesn't create a copy, just a reference. Was `df.copy()` intended here?